timApp/static/scripts/tim/item/relevance-edit.component.ts

```diff
@@ -75,64 +86,104 @@ export class RelevanceEditComponent {
-            this.isDefault = r.result.data.default;
-            this.isInherited = r.result.data.inherited;
-            this.relevance = r.result.data.relevance.relevance;
+            this.isDefault = r.result.default;
+            this.isInherited = r.result.inherited;
+            this.relevance = r.result.relevance.relevance;
         } else {
-            this.errorMessage = r.result.data.error;
+            this.errorMessage = r.result.error.error;
         }
     }
 
     /**
      * Set new relevance value for the item.
+     * @param itemID id number for the document/item
      * @param newValue Input value.
      */
-    private async setRelevance(newValue: number) {
+    private async setRelevance(itemID: number, newValue: number) {
         this.errorMessage = undefined;
-        const r = await to(
-            $http.post(`/items/relevance/set/${this.item.id}`, {
+        const r = await toPromise(
+            this.http.post(`/items/relevance/set/` + itemID, {
                 value: newValue,
             })
         );
         if (r.ok) {
             this.isDefault = false;
             this.isInherited = false;
         } else {
-            this.errorMessage = r.result.data.error;
+            this.errorMessage = r.result.error.error;
         }
     }
 
     /**
      * Return default (or inherited) relevance value.
+     * @param itemID id number for the document/item
      */
-    private async resetRelevance() {
+    private async resetRelevance(itemID: number) {
         this.errorMessage = undefined;
-        const r = await to(
-            $http.get<IRelevance>(`/items/relevance/reset/${this.item.id}`)
+        const r = await toPromise(
+            this.http.get<IRelevance>(`/items/relevance/reset/` + itemID)
         );
         if (!r.ok) {
-            this.errorMessage = r.result.data.error;
+            this.errorMessage = r.result.error.error;
         }
         void this.getRelevance();
+
+        if (this.updateTranslations && this.relevance) {
+            await this.updateTranslationRelevances(this.relevance);
+        }
     }
 
     async resetClicked() {
-        await this.resetRelevance();
+        await this.resetRelevance(this.item.id);
     }
 
     async saveClicked() {
         if (this.relevance != null) {
-            await this.setRelevance(this.relevance);
+            await this.setRelevance(this.item.id, this.relevance);
+            if (this.updateTranslations) {
+                await this.updateTranslationRelevances(this.relevance);
+            }
         } else {
             this.errorMessage =
                 "Incorrect relevance value: input a whole number!";
         }
     }
+
+    /**
+     * Update the relevance values for translations based on the current document
+     * @param sourceRelevance relevance value for the source document
+     * @private
+     */
+    private async updateTranslationRelevances(sourceRelevance: number) {
```

Comment:
Mietin, pitäisikö tämän olla toteutettu palvelimen puolella? Eli se `set_relevance` ottaisi jonkun parametrin mallia `update_translations`, joka päivittäisi käännösten relevanssit palvelimella.

Käytännön merkitystä olisi pääosin, että jos esim CLIlla muuttaa relevanssin, niin se automaattisesti osaa päivittää sen oikein käännöksille.

Reply:
Joo totta, minä muuttelen niin päin.